### PR TITLE
661 contract  event filtering

### PR DESCRIPTION
**Pull request by Valazan**
# Refactor: Contract Proxies

## Overview
This PR introduces a series of enhancements and tests focused on contract proxies. The key features include the addition of a contract filter, improved event filtering mechanisms, and comprehensive documentation on event filtering examples.

Example of an event filtering:

```
// Get the transfer events for a specific contract
const transferEvents = await contract.filters.Transfer().get();

```
## Changes
- **[feat]**: Added a contract filter to enhance contract interaction capabilities.
- **[test]**: Implemented tests for filtering transfer events to ensure robustness in event handling.
- **[docs]**: Provided detailed documentation on event filtering, including examples to guide users.

## Commit Summary
- Added contract filter functionality.
- Added multiple tests for filtering transfer events.
- Added documentation for event filtering and a specific example for filtering four-argument events.

### CODE REVIEW

File: packages/core/src/abi/fragment.ts [hunk 1]
```diff
@@ -178,7 +179,7 @@ class Event<ABIType> {
     /**
      * The main fragment handled by ethers.js.
      */
-    public fragment: ethers.EventFragment;
+    public fragment: EventFragment;
```
Comment (fabiorigam) [inline packages/core/src/abi/fragment.ts:183]: There is another ethers.EventFragment to substitute on this page

**Review by rodolfopietro97 — CHANGES_REQUESTED** [review]
@Valazan Is it tested over different methods or it is only for transfer (I give an on the fly look)

**Comment by Valazan** [comment]
@rodolfopietro97 I've just added two additional contracts (one of them is the testing contract)

File: docs/templates/contracts.md [hunk 2]
```diff
@@ -81,3 +81,13 @@ Once the contract is compiled, we can deploy it using the vechain SDK. The follo
 Once the contract is deployed, we can transfer tokens to another address using the vechain SDK. The following code shows how to transfer 10000 token smallest unit to another address:
 
 [ERC20FunctionCallSnippet](examples/contracts/contract-transfer-ERC20-token.ts)
+
+
+#### Filter the Transfer event
+
+
+Once the contract is deployed, we can filter the Transfer event using the vechain SDK. The following code shows how to filter the Transfer event:
```
Comment (fabiorigam) [inline docs/templates/contracts.md:89]: Can we add more details? We need to improve the docs

File: packages/network/src/thor-client/contracts/model/contract.ts [hunk 3]
```diff
@@ -214,6 +164,26 @@ class Contract {
         }
         return functionFragment;
     }
+
+    /**
+     * Retrieves the event fragment for the specified event name.
+     * @param eventName - The name of the event.
+     */
```
Comment (fabiorigam) [inline packages/network/src/thor-client/contracts/model/contract.ts:171]: Missing @returns

**Review by fabiorigam — APPROVED** [review]

**Review by rodolfopietro97 — APPROVED** [review]

**Pull request merged**